### PR TITLE
[CI_ONLY] S3 blob store CI instabilities 2

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/blob/blob-s3/pom.xml [hunk 1]
```diff
@@ -82,6 +82,10 @@
             <groupId>io.projectreactor.addons</groupId>
             <artifactId>reactor-extra</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.projectreactor.addons</groupId>
+            <artifactId>reactor-pool</artifactId>
```
Comment (chibenwa) [inline server/blob/blob-s3/pom.xml:87]: This looks cool!

Maybe we want to evaluate the usage of this pool along with the rabbitMQ driver....

File: server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3DumbBlobStore.java [hunk 2]
```diff
@@ -94,12 +97,21 @@
...
             .build();
+
+        clientPool = PoolBuilder.from(Mono.fromCallable(clientCreator))
+            .acquisitionScheduler(Schedulers.elastic())
+            .destroyHandler(client -> Mono.fromRunnable(client::close))
+            .maxPendingAcquireUnbounded()
+            .sizeBetween(4, 64)
+            .fifo();
+
+        clientPool.warmup().block();
```
Comment (chibenwa) [inline server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3DumbBlobStore.java:108]: The warmup should not be done in a constructor. We should rather, if adopted, follow the Startable patern.

File: server/blob/blob-s3/pom.xml (hunk 1 above)
Comment (blackheaven) [reply server/blob/blob-s3/pom.xml:87]: sadly we cannot since the `Channel` leaks, which is not the case here

Comment (chibenwa) [reply server/blob/blob-s3/pom.xml:87]: I'm always impressed at how bad the java drivers for RabbitMQ are...

**Comment by chibenwa** [comment]
mvn tests :green_apple:

Fails in deployment tests

```
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:41
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] LastClientMsg: 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Expected: 'A002 OK (\[.+\] )?APPEND completed.'
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Actual   : 'A002 NO APPEND processing failed.'
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR] validateDeploymentWithMailsFromSmtp  Time elapsed: 0.637 s  <<< ERROR!
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] org.apache.james.utils.SMTPSendingException: 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Error upon step Data: 451 Unable to process request
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR] selectThenFetchWithExistingMessages  Time elapsed: 0.792 s  <<< ERROR!
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] org.apache.james.mpt.protocol.ProtocolSession$InvalidServerResponseException: 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Location: /org/apache/james/imap/scripts/SelectThenFetchWithExistingMessages.test:42
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] LastClientMsg: Test content
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Expected: '\* 1 EXISTS'
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] Actual   : 'A04 NO APPEND processing failed.'
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [INFO] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [INFO] Results:
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [INFO] 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR] Errors: 
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.selectThenFetchWithExistingMessages:86 » InvalidServerResponse
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeployment:81 » InvalidServerResponse
[bfe79bc04a5485c9a7917ac21aba4cd382f0cc31] [ERROR]   JamesDeploymentValidationTest>DeploymentValidation.validateDeploymentWithMailsFromSmtp:92 » SMTPSending
```

Configuration issue?

`Running integration tests on Cassandra + RabbitMQ + object store on s3-blob-store-rebase-2-instabilities-2-ebfcff05-c761-43b2-b449-395367d02fde`

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/3773

**Pull request closed without merging**